lib/clientv2/bgpconfig.go

```
@@ -0,0 +1,107 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv2
+
+import (
+	"context"
+	"errors"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	"github.com/projectcalico/libcalico-go/lib/watch"
+)
+
+// BGPConfigurationInterface has methods to work with BGPConfiguration resources.
+type BGPConfigurationInterface interface {
+	Create(ctx context.Context, res *apiv2.BGPConfiguration, opts options.SetOptions) (*apiv2.BGPConfiguration, error)
+	Update(ctx context.Context, res *apiv2.BGPConfiguration, opts options.SetOptions) (*apiv2.BGPConfiguration, error)
+	Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.BGPConfiguration, error)
+	Get(ctx context.Context, name string, opts options.GetOptions) (*apiv2.BGPConfiguration, error)
+	List(ctx context.Context, opts options.ListOptions) (*apiv2.BGPConfigurationList, error)
+	Watch(ctx context.Context, opts options.ListOptions) (watch.Interface, error)
+}
+
+// bgpConfigurations implements BGPConfigurationInterface
+type bgpConfigurations struct {
+	client client
+}
+
+// Create takes the representation of a BGpConfiguration and creates it.
+// Returns the stored representation of the BGPConfiguration, and an error
+// if there is any.
+func (r bgpConfigurations) Create(ctx context.Context, res *apiv2.BGPConfiguration, opts options.SetOptions) (*apiv2.BGPConfiguration, error) {
+	if res.ObjectMeta.GetName() != "default" && (res.Spec.NodeToNodeMeshEnabled != nil || res.Spec.DefaultNodeASNumber != nil) {
+		return nil, errors.New("Cannot set spec fields node_to_node_mesh_enabled and default_node_as_number on a non default BGP Configuration.")
```

You should return a validation error (defined in the errors package).